File: test-suite/js/security/hardcoded-secrets-clean.ts
function requiredEnv(name: string): string {
  const value = process.env[name];
  if (!value) {
    throw new Error(`Missing ${name}`);
  }
  return value;
}

function loadRefreshToken(): string {
  return requiredEnv("REFRESH_TOKEN");
}

export const NEXTAUTH_SECRET = requiredEnv("NEXTAUTH_SECRET");

const stripeSecretKey = process.env.STRIPE_SECRET_KEY!;

export const mapsApiKey = requiredEnv("MAPS_API_KEY");

const authConfig = {
  jwtSecret: requiredEnv("JWT_SECRET"),
  clientSecret: requiredEnv("OAUTH_CLIENT_SECRET"),
  session: {
    secret: requiredEnv("SESSION_SECRET"),
  },
};

export const webhookConfig = {
  signingSecret: requiredEnv("STRIPE_WEBHOOK_SECRET"),
  refreshToken: loadRefreshToken(),
};

export const displayConfig = {
  themeVariant: process.env.THEME_VARIANT || "dark",
  publicBaseUrl: process.env.NEXT_PUBLIC_APP_URL || "https://example.com",
};

export { authConfig, stripeSecretKey };

File: test-suite/python/security/hardcoded_secrets_clean.py
import os
from os import environ, getenv


def required_env(name: str) -> str:
    value = os.environ[name]
    if not value:
        raise RuntimeError(f"Missing {name}")
    return value


SECRET_KEY = required_env("SECRET_KEY")
STRIPE_SECRET_KEY: str = required_env("STRIPE_SECRET_KEY")

auth_config = {
    "JWT_SECRET": required_env("JWT_SECRET"),
    "OAUTH_CLIENT_SECRET": required_env("OAUTH_CLIENT_SECRET"),
    "SESSION_SECRET": os.environ["SESSION_SECRET"],
    "WEBHOOK_SECRET": getenv("WEBHOOK_SECRET"),
}

settings = {}
settings["API_KEY"] = environ["API_KEY"]
settings["SIGNING_SECRET"] = required_env("SIGNING_SECRET")
settings.setdefault("COOKIE_SECRET", required_env("COOKIE_SECRET"))

flask_mapping = dict(SECRET_KEY=required_env("FLASK_SECRET_KEY"))


def jwt_secret() -> str:
    return required_env("JWT_SECRET")


def issue_token(access_token: str = required_env("ACCESS_TOKEN")) -> str:
    return access_token


def build_callback(retry_count: int = 3) -> int:
    return retry_count


class ProviderConfig:
    clientSecret = required_env("PROVIDER_CLIENT_SECRET")


apiKey = required_env("MAPS_API_KEY")
display_config = {
    "THEME_VARIANT": os.getenv("THEME_VARIANT", "dark"),
    "PUBLIC_BASE_URL": os.getenv("PUBLIC_BASE_URL", "https://example.com"),
}


# Issue #64 regressions: secret-ish names describing non-secret metadata.
oauth_client = dict(
    token_uri="https://oauth2.googleapis.com/token",
    auth_uri="https://accounts.google.com/o/oauth2/auth",
)

token_metadata = {
    # Prose under a secret-ish key is not a credential literal.
    "token": "issued after the OAuth flow completes",
    "token_url": "https://example.org/oauth/token",
}
